presentation/src/main/java/org/cryptomator/presentation/util/BiometricAuthenticationMigration.kt

```diff
@@ -0,0 +1,142 @@
+	) {
+		Timber.tag("BiometricAuthMigration").d("Prompt for decryption")
+		handleBiometricAuthentication(fragment = fragment, cryptoMode = CryptoMode.CBC, password = vaultModel.password!!, allVaults = allVaults, onSuccess = { decryptedPassword ->
+			decryptedVaults.add(
+				VaultModel(
+					vault = Vault.aCopyOf(vaultModel.toVault()).withSavedPassword(decryptedPassword, CryptoMode.NONE).build()
+				)
+			)
+			processNextVault(fragment, vaultQueue, decryptedVaults, reEncryptedVaults, allVaults)
+		})
+	}
+
+	private fun encryptUsingGcm(
+		fragment: Fragment, vaultModel: VaultModel, vaultQueue: ArrayDeque<VaultModel>, decryptedVaults: MutableList<VaultModel>, reEncryptedVaults: MutableList<VaultModel>, allVaults: List<VaultModel>
+	) {
+		Timber.tag("BiometricAuthMigration").d("Prompt for encryption")
+		handleBiometricAuthentication(fragment = fragment, cryptoMode = CryptoMode.GCM, password = vaultModel.password!!, allVaults = allVaults, onSuccess = { encryptedPassword ->
+			reEncryptedVaults.add(
+				VaultModel(
+					vault = Vault.aCopyOf(vaultModel.toVault()).withSavedPassword(encryptedPassword, CryptoMode.GCM).build()
+				)
+			)
+			processNextVault(fragment, vaultQueue, decryptedVaults, reEncryptedVaults, allVaults)
+		})
+	}
+
+	private fun handleBiometricAuthentication(
+		fragment: Fragment, cryptoMode: CryptoMode, password: String, allVaults: List<VaultModel>, onSuccess: (String) -> Unit
+	) {
+		try {
+			val biometricAuthCryptor = BiometricAuthCryptor.getInstance(context, cryptoMode)
+			val authCallback = object : BiometricPrompt.AuthenticationCallback() {
+				override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+					super.onAuthenticationSucceeded(result)
+					Timber.tag("BiometricAuthMigration").d("Authentication succeeded")
+					val cipher = result.cryptoObject?.cipher
+					try {
+						val processedPassword = when (cryptoMode) {
+							CryptoMode.CBC -> biometricAuthCryptor.decrypt(cipher, password)
+							CryptoMode.GCM -> biometricAuthCryptor.encrypt(cipher, password)
+							CryptoMode.NONE -> throw IllegalStateException("CryptoMode.NONE is not allowed here")
+						}
+						onSuccess(processedPassword)
+					} catch (e: BadPaddingException) {
+						Timber.e(e, "BadPaddingException - possibly due to an invalidated key")
+						callback.onBiometricKeyInvalidated(allVaults)
+					}
+				}
+
+				override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+					super.onAuthenticationError(errorCode, errString)
+					Timber.e("Authentication error: %s errorCode=%d", errString, errorCode)
+					callback.onBiometricAuthenticationFailed(allVaults)
+				}
+
+				override fun onAuthenticationFailed() {
+					super.onAuthenticationFailed()
+					Timber.e("Authentication failed")
+				}
```

Comment:
Because we do not want to drop the vault passwords only because you show the face of your cat into your camera or the phone simply does not recognize it.